composer/optim/scheduler.py

```diff
@@ -68,137 +64,110 @@ def _convert_time(time: Union[str, Time[int], Time[float]], state: State, ssr: f
     return Time(value=int(time.value * ssr), unit=time.unit)
 
 
-def compile(scheduler: ComposerScheduler, state: State) -> Scheduler:
-
-    if isinstance(scheduler, Scheduler):
-        return scheduler
-
+def compile_composer_scheduler(scheduler: ComposerScheduler, state: State, ssr: float = 1.0) -> LambdaLR:
     optimizers = state.optimizers
     if len(optimizers) != 1:
         raise NotImplementedError("Providing functional schedulers is unsupported with multiple optimizers.")
     optimizer = optimizers[0]
 
     def scheduler_fn(epoch: int) -> float:
-        del epoch  # unused
-        return scheduler(state)
+        del epoch  # unused. Provided by the pytorch LambdaLR
+        return scheduler(state, ssr)
 
     lambda_scheduler = LambdaLR(optimizer, scheduler_fn)
 
     return lambda_scheduler
 
 
-def step_scheduler(state: State, *, ssr: float = 1.0, step_size: Union[str, Time], gamma: float = 0.1) -> float:
+class StepScheduler:
```

Comment:
We do not re-document base class docstrings. I purposefully left out the inheritance because Protocols need not be inherited and wanted to (implicitly) ensure that we do not use any `assert isinstance(scheduler, ComposerScheduler)`, since that would fail if the user provided a lambda instead of a ComposerScheduler object.

Reply:
Yeah, I agree that it's important that `ComposerScheduler` stay as a `Protocol`, but the explicit inheritance will make the docs clearer - right now the Sphinx docs describe `CosineAnnealingScheduler`, for instance, as basing `object`, not `ComposerScheduler`.

Reply:
Re: base of object, that's a sphinx issue we need to fix -- it should be possible to hide that (this happens everywhere and it's kinda annoying).
Agree that it would be nice of having a base of ComposerScheduler. I will change that and add a test case.